addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,476 @@
+                                        this.thing.getBridgeUID(), event.getAddr());
+                                if (event.getAddr().toString()
+                                        .equalsIgnoreCase((String) (configuration.get("ds18b20_address")))) {
+                                    updateState(channelId, new QuantityType<Temperature>(
+                                            Double.parseDouble(event.getTemp()), SIUnits.CELSIUS));
+                                } else {
+                                    logger.debug("The address of {} does not match {} not updating this channel",
+                                            event.getAddr().toString(), (configuration.get("ds18b20_address")));
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    logger.warn(
+                            "The auth token: {}  did not match {} configured in the thing {} so the state was not accepted.",
+                            sentAuth, config.authToken, getThing().getThingTypeUID().toString());
+                }
+            } else {
+                // by logging all of the ones that don't match in debug we can see all of the channel ids that have been
+                // defined
+                logger.debug("The channel sent by the module: {} did not match channelId: {} for thing {}", sentZone,
+                        channelId, getThing().getThingTypeUID().toString());
+            }
+        });
+    }
+
+    private void checkConfigruation() throws ConfigValidationException {
+        KonnectedConfiguration testConfig = getConfigAs(KonnectedConfiguration.class);
+
+        if ((testConfig.hostAddress == null) && (callbackIpAddress == null)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Unable to obtain hostaddress from OSGI service, please manually configure hostaddress");
+        }
+
+        else {
+            this.config = testConfig;
+        }
+
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters)
+            throws ConfigValidationException {
+
+        this.validateConfigurationParameters(configurationParameters);
+        for (Entry<String, Object> configurationParameter : configurationParameters.entrySet()) {
+            Object value = configurationParameter.getValue();
+            logger.debug("Controller Configuration update {} to {}", configurationParameter.getKey(), value);
+
+            if (value == null) {
+                continue;
+            }
+
+            String[] cfg = configurationParameter.getKey().split("_");
+            if ("controller".equals(cfg[0])) {
+                if (cfg[1].equals("softreset") && value instanceof Boolean && ((Boolean) value) == true) {
+                    try {
+                        http.doGet(moduleIpAddress + "/settings?restart=true");
```

Comment:
Afaik, this method has to return quickly and not block - http calls are potentially long running, so they should not be done here. Anyhow, if you model this as a channel, this problem will be gone.

Reply:
Yeah, that is an ugly hack in the ZWave binding, which should not serve as an example...
But yes, as you have it all in place like that now and it is probably the easier migration towards eclipse/smarthome#6398, you could put such calls into an async job and we live with that for the time being.

Reply:
Correction: I actually thought you were referencing https://github.com/eclipse/smarthome/issues/3484 - I think it is in the scope of this issue to also address such features as resets.

Reply:
placed all three commands in the scheduler to make them an async call